tensorboard/webapp/metrics/views/card_renderer/utils_test.ts

```diff
@@ -310,4 +311,140 @@ describe('metrics card_renderer utils test', () => {
       expect(getClosestStep(10, [0, 10, 20])).toBe(10);
     });
   });
+
+  describe('#maybeClipLinkedTimeSelection', () => {
+    it('changes start when start.step < minStep', () => {
+      expect(
+        maybeClipLinkedTimeSelection(
+          {
+            start: {step: 0},
+            end: null,
+          },
+          1,
+          4
+        )
+      ).toEqual({
+        startStep: 1,
+        endStep: null,
+        clipped: true,
+      });
+    });
+
+    it('changes end when end.step < maxStep', () => {
+      expect(
+        maybeClipLinkedTimeSelection(
+          {
+            start: {step: 0},
+            end: {step: 4},
+          },
+          0,
+          3
+        )
+      ).toEqual({
+        startStep: 0,
+        endStep: 3,
+        clipped: true,
+      });
+    });
+
+    it('does not create an endstep when timeSelection does not contain one', () => {
+      expect(
+        maybeClipLinkedTimeSelection(
+          {
+            start: {step: 10},
+            end: null,
+          },
+          0,
+          20
+        )
+      ).toEqual({
+        startStep: 10,
+        endStep: null,
+        clipped: false,
+      });
+    });
+
+    it('returns minStep and maxStep when the timeselection is a superset of the min/maxstep', () => {
+      expect(
+        maybeClipLinkedTimeSelection(
+          {
+            start: {step: 0},
+            end: {step: 100},
+          },
+          30,
+          50
+        )
+      ).toEqual({
+        startStep: 30,
+        endStep: 50,
+        clipped: true,
+      });
+    });
+
+    it('returns startStep === endStep === maxStep when timeSelection is above maxStep', () => {
```

Comment:
"startStep === endStep === maxStep" is not meaningful to me. I think what you try to convey is "clips both fob to maxStep"?